extensions/amp-story-interactive/1.0/utils.js

```diff
@@ -15,11 +15,48 @@
  */
 
 import {Action} from '../../amp-story/1.0/amp-story-store-service';
+import {map} from '../../../src/utils/object';
+import {toArray} from '../../../src/types';
+import {user} from '../../../src/log';
 
 export const updateInteractiveStoreState = (storeService, interactive) => {
   const update = {
     'option': interactive.getOptionSelected(),
-    'interactiveId': this.getInteractiveId_(),
+    'interactiveId': interactive.getInteractiveId(),
   };
   storeService.dispatch(Action.ADD_INTERACTIVE_REACT, update);
 };
+
+/** @private Whether ids are deduplicated or not */
+const deduplicatedIds = false;
+
+/**
+ * Deduplicates the interactive Ids, only called once
+ * @param {Element} doc
+ */
+export const deduplicateInteractiveIds = (doc) => {
+  if (deduplicatedIds) {
+    return;
+  }
+  deduplicatedIds = true;
+  const interactiveEls = doc.querySelectorAll(
+    'amp-story-interactive-binary-poll, amp-story-interactive-poll, amp-story-interactive-quiz'
+  );
+  const interactiveIds = toArray(interactiveEls).map(
+    (el) => el.id || 'interactive-id'
+  );
```

Comment:
Nit: you don't need this extra loop, it could be done in the next `for`

Reply:
True, I was following amp-story.js that has this same logic. Might be a good idea to do this in initializePageIds at some point

Reply:
initializePageIds uses the pageIds to register them in the store so it needs this extra step